dfa/duration_extraction.py

```diff
@@ -49,13 +49,18 @@ def to_adj_matrix(mat):
     return adj_mat.tocsr()
 
 
-def extract_durations_with_dijkstra(seq: np.array, pred_matrix: np.array, tokenizer) -> np.array:
+def extract_durations_with_dijkstra(tokens: np.array, pred: np.array) -> np.array:
     """
     Extracts durations from the attention matrix by finding the shortest monotonic path from
     top left to bottom right.
     """
 
-    path_probs = 1. - pred_matrix
+    target_len, pred_len = tokens.shape[0], pred.shape[0]
+    pred_max = np.zeros((pred_len, target_len))
+    for i in range(pred.shape[0]):
+        pred_max[i] = pred[i, tokens]
+
+    path_probs = 1. - pred_max
```

Comment:
I'm confused, why are you now only assigning one value per row?

Reply:
Here it calculates for each mel step i the probabilities for all tokens in the sequence. The matrix is then of dim (mel steps, sequence steps) wehre entries i,j are the prediction probs for token j in the sequence at mel step i.